src/actions/sagemaker/sagemaker_infer.ts

```diff
@@ -0,0 +1,346 @@
+      description: "Port for sending emails.",
+    },
+    {
+      name: "smtpFrom",
+      label: "SMTP From",
+      required: true,
+      sensitive: false,
+      description: "From for sending emails.",
+    },
+    {
+      name: "smtpUser",
+      label: "SMTP User",
+      required: true,
+      sensitive: false,
+      description: "User for sending emails.",
+    },
+    {
+      name: "smtpPass",
+      label: "SMTP Pass",
+      required: true,
+      sensitive: false,
+      description: "Pass for sending emails.",
+    },
+  ]
+
+  async execute(request: Hub.ActionRequest) {
+
+    try {
+      // get string inputs
+      const {
+        modelName,
+        bucket,
+        awsInstanceType,
+      } = request.formParams
+
+      const { roleArn } = request.params
+
+      // validate string inputs
+      if (!modelName) {
+        throw new Error("Need SageMaker model name.")
+      }
+      if (!bucket) {
+        throw new Error("Need Amazon S3 bucket.")
+      }
+      if (!awsInstanceType) {
+        throw new Error("Need Amazon awsInstanceType.")
+      }
+      if (!roleArn) {
+        throw new Error("Need Amazon Role ARN for SageMaker & S3 Access.")
+      }
+
+      const numInstances = this.getNumericFormParam(request, "numInstances", 1, 500)
+      const numStripColumns = this.getNumericFormParam(request, "numStripColumns", 0, 2)
+
+      // upload data to S3
+      const date = Date.now()
+      const jobName = `transform-job-${date}`
+      const inputDataKey = `${jobName}/transform-input`
+      const rawDataKey = `${jobName}/transform-raw`
+
+      // store data in S3 bucket
+      await this.uploadToS3(request, bucket, numStripColumns, inputDataKey, rawDataKey)
+
+      // create transform job
+      const client = this.getSageMakerClientFromRequest(request)
+
+      const s3InputPath = `s3://${bucket}/${inputDataKey}`
+      const s3OutputPath = `s3://${bucket}`
+      winston.debug("s3InputPath", s3InputPath)
+      winston.debug("s3OutputPath", s3OutputPath)
+
+      const transformParams: SageMaker.CreateTransformJobRequest = {
+        ModelName: modelName,
+        TransformJobName: jobName,
+        TransformInput: {
+          DataSource: {
+            S3DataSource: {
+              S3DataType: "S3Prefix", // required
+              S3Uri: s3InputPath, // required
+            },
+          },
+          ContentType: "text/csv",
+        },
+        TransformOutput: {
+          S3OutputPath: s3OutputPath,
+        },
+        TransformResources: {
+          InstanceCount: numInstances,
+          InstanceType: awsInstanceType,
+        },
+      }
+      winston.debug("transformParams", transformParams)
+
+      const transformResponse = await client.createTransformJob(transformParams).promise()
+      winston.debug("transformResponse", transformResponse)
+
+      // return success response
+      return new Hub.ActionResponse({ success: true })
+
+    } catch (err) {
+      winston.error(err)
```

Comment:
For anything that is going to be logged at error or higher - the ` request.webhookId ` is a nice thing to add for debugging. That way we can track when and how a request was made in Looker and if there were any errors constructing the it